### PR TITLE
api: resolve json references

### PR DESCRIPTION
**Pull request by Theer108**
Work in progress, but it works fine.

### CODE REVIEW

**Comment by jirikuncar** [comment]
@Theer108 please remove all CDS mentions.

File: invenio_records/utils.py [hunk 1]
```diff
@@ -24,14 +24,49 @@
...
+
+    def __init__(self, *args, **kwargs):
+        super(CDSJsonLoader, self).__init__(*args, **kwargs)
+        self.resolve_map = {
+            'record': self.resolve_record
+        }
+
+    def get_remote_json(self, uri, **kwargs):
+
+        splitted_url = urlparse.urlsplit(uri)
```
Comment (jirikuncar) [inline invenio_records/utils.py]: We can take an advantage of Werkzeug routing system. Another nice thing would be to use *Flask-Registry* to collect JSON reference loaders.

```python
from werkzeug.routing import Map, Rule
url_map = Map([Rule('/record/<id>', endpoint='cds_record', host='cds.cern.ch'), Rule('/record/<id>', endpoint='zenodo_record', host='zenodo.cern.ch')], host_matching=True)
url_map.bind('cds.cern.ch').match('/record/1')
```

**Comment by Theer108** [comment]
Depends on invenio-jsonref

File: setup.py [hunk 2]
```diff
@@ -40,6 +40,7 @@
     # FIXME 'Invenio>2.1',
     'dojson>=0.1.0',
     'invenio-pidstore>=0.1.0',  # TODO consider making it optional
+    'invenio-jsonref>=0.1.0'
```
Comment (Theer108) [inline setup.py]: Should it be here as dependency or we can omit it if it is already in invenio?

Comment (jirikuncar) [inline setup.py]: @Theer108 it should be here :+1:

Comment (jirikuncar) [inline setup.py]: please add `,` comma at the end of line. Thanks

**Comment by kaplun** [comment]
@Theer108 you can git-squash these 3 commits as they are adding and amending the same functionality.

**Comment by Theer108** [comment]
@kaplun Done

**Comment by tiborsimko** [comment]
Oftentimes, one wants to get the raw JSON representing the record out quickly, with all the `$ref` in place "as is" -- e.g. for bibliographic record management purposes.  We may want to make dereferencing optional, the default getter wouldn't do it, and plug dereferencing into PostgreSQL->Elasticsearch JSON enriching step only. See also musings at https://github.com/inveniosoftware/invenio-authorities/issues/1.

**Comment by lnielsen** [comment]
Superseded by #69

**Pull request closed without merging**